### ISSUE TITLE
Select: Accessibility improvements

### ISSUE THREAD

**Issue by tomheller** [dynatrace-oss/barista#568]
# Feature Request

## Summary

The barista select currently fails some accessibility tests, as it does not conform to every aspect of the [listbox role](https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-scrollable.html).
We may as well think about moving the role to a [combobox](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html) one, as this seems to also fit.

We should look into connecting the combo box with the autocompletes via the attributes:

* `aria-expanded`
* `aria-owns`
* `aria-haspopup`

**Comment by tomheller** [comment]
After some investigation, we have reached the conclusion that this will be very hard to achieve with the current select setup. 
The biggest problem is that the options and the options wrapper do not exist in the DOM until they are visible on the page (usually when focusing the select). This is due to the fact that we are using a cdkOverlay to show and position the options of a select. 

Switching to a `combobox` role also isn't right, because for that the select would need to contain a _text-input_ element to filter option values. 

The `listbox` does not let you connect an non-child component as a _option-wrapper_. 
Our best bet to improve accessibility of the select is to completely rethink the structure of the elements, when redesigning this one for the next version.
